src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/command/flow/ingress/IngressFlowSegmentBase.java

```diff
@@ -153,11 +194,19 @@ protected void setup(FloodlightModuleContext moduleContext) throws Exception {
         return commandProcessor.chain(meterDryRun);
     }
 
-    private CompletableFuture<FlowSegmentReport> planOfFlowsInstall(MeterId effectiveMeterId) {
+    private CompletableFuture<FlowSegmentReport> planOfFlowsInstall(EffectiveIds effectiveIds) {
+        MeterId effectiveMeterId = effectiveIds.getMeterId();
         if (effectiveMeterId == null && rulesContext != null && !rulesContext.isUpdateMeter()) {
             effectiveMeterId = getMeterConfig().getId();
         }
-        List<OFFlowMod> ofMessages = makeFlowModMessages(effectiveMeterId);
+
+        GroupId effectiveGroupId = effectiveIds.getGroupId();
+        MirrorConfig mirrorConfig = getMirrorConfig();
+        if (effectiveGroupId == null && mirrorConfig != null) {
+            effectiveGroupId = mirrorConfig.getGroupId();
```

Comment:
It looks weird. Add at least comment, why this have happend and why we should ignore `effectiveGroupId == null`.

Reply:
This check has been removed